File: app/src/main/java/com/commit451/gitlab/model/api/RepositoryCommit.java
package com.commit451.gitlab.model.api;

import com.commit451.gitlab.util.ObjectUtil;
import com.squareup.moshi.Json;

import org.parceler.Parcel;

import java.util.Date;

@Parcel
public class RepositoryCommit {
    @Json(name = "id")
    String id;
    @Json(name = "short_id")
    String shortId;
    @Json(name = "title")
    String title;
    @Json(name = "author_name")
    String authorName;
    @Json(name = "author_email")
    String authorEmail;
    @Json(name = "created_at")
    Date createdAt;
    @Json(name = "message")
    String message;

    public RepositoryCommit() {}

    public String getId() {
        return id;
    }

    public String getShortId() {
        return shortId;
    }

    public String getTitle() {
        return title;
    }

    public String getAuthorName() {
        return authorName;
    }

    public String getAuthorEmail() {
        return authorEmail;
    }

    public Date getCreatedAt() {
        return createdAt;
    }

    public String getMessage() {
        return message;
    }

    @Override
    public boolean equals(Object o) {
        if (!(o instanceof RepositoryCommit)) {
            return false;
        }

        RepositoryCommit commit = (RepositoryCommit) o;
        return ObjectUtil.INSTANCE.equals(id, commit.id);
    }

    @Override
    public int hashCode() {
        return ObjectUtil.INSTANCE.hash(id);
    }
}

File: app/src/main/java/com/commit451/gitlab/model/api/RepositoryCommit.kt
package com.commit451.gitlab.model.api

import android.os.Parcelable
import com.squareup.moshi.Json
import kotlinx.android.parcel.Parcelize
import java.util.Date

@Parcelize
data class RepositoryCommit(
    @Json(name = "id")
    var id: String,
    @Json(name = "short_id")
    var shortId: String? = null,
    @Json(name = "title")
    var title: String? = null,
    @Json(name = "author_name")
    var authorName: String? = null,
    @Json(name = "author_email")
    var authorEmail: String? = null,
    @Json(name = "created_at")
    var createdAt: Date? = null,
    @Json(name = "message")
    var message: String? = null
) : Parcelable
